configuration_go/k8sutil/genericdeployment.go

```diff
@@ -0,0 +1,155 @@
+package k8sutil
+
+import (
+	monv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// SidecarConfig represents the configuration required to add extra containers
+// to a particular Deployment/StatefulSet.
+type SidecarConfig struct {
+	Sidecars                      []corev1.Container
+	AdditionalPodVolumes          []corev1.Volume
+	AdditionalServicePorts        []corev1.ServicePort
+	AdditionalServiceMonitorPorts []monv1.Endpoint
+}
+
+// GenericDeploymentConfig represents certain config fields
+// that might be useful to add/override in a Deployment. It contains
+// fields of both DeploymentSpec and PodSpec.
+// It also has method defined for overriding any default values.
+type GenericDeploymentConfig struct {
```

Comment:
`DeploymentConfig` was its own thing in OpenShift for a long time. Do you think the naming here could become any source of confusion in that respect?

Reply:
Ah, I wasn't aware..let's rename then

Reply:
Have renamed it.